core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -466,7 +468,22 @@ private void notifyEndOfRebalanceIfNeeded(int topologyId, int rebalanceId) {
             log.debugf("Finished receiving of segments for cache %s for topology %d.", cacheName, topologyId);
             stopApplyingState();
             stateTransferManager.notifyEndOfRebalance(topologyId, rebalanceId);
-            stateRequestCompletionService.drainCompletionQueue();
+            List<? extends Future<Void>> futures = stateRequestCompletionService.drainCompletionQueue();
+            boolean interrupted = false;
+            for (Future<Void> future : futures) {
+               // These will not block since they are already completed
+               try {
+                  future.get();
+               } catch (InterruptedException e) {
+                  // This shouldn't happen as each task is complete
+                  interrupted = true;
+               } catch (ExecutionException e) {
+                  log.topologyUpdateError(topologyId, e.getCause());
```

Comment:
@wburns Isn't this a bit late to log the errors? I mean if there was a problem requesting state, the state transfer may never end and the queue will never be drained.

I would rather add an uncaught exception handler in `DefaultThreadFactory` to log the error instead of just writing it to `System.err`.